### ISSUE TITLE
wrong reported log for DIGITALTWIN_CLIENT_RESULT_INVALID and QOS_VALUE_INVALID 

### ISSUE THREAD

**Issue by PezzoniL** [Azure/azure-iot-sdk-c#1285]
I had cloned the latest public-preview for testing IoTCentral PnP
Looking the latest version there C-SDK there is the introduction of:

**MU_DEFINE_ENUM_WITHOUT_INVALID(DIGITALTWIN_CLIENT_RESULT, DIGITALTWIN_CLIENT_RESULT_VALUES);**

instead of 
MU_DEFINE_ENUM(DIGITALTWIN_CLIENT_RESULT, DIGITALTWIN_CLIENT_RESULT_VALUES);

and the introduction of 
**MU_DEFINE_ENUM_2(QOS_VALUE, QOS_VALUE_VALUES)**
Instead of:
MU_DEFINE_ENUM(QOS_VALUE, QOS_VALUE_VALUES)

These modifications produce the following debug console log:

>Info: DeviceInfoInterface_InterfaceRegisteredCallback with status=><**DIGITALTWIN_CLIENT_RESULT_INVALID**>, userContext=<20003458>
Info: DEVICEINFO_INTERFACE: Interface successfully registered.

and:
><- 11:20:38 PUBLISH | IS_DUP: false | RETAIN: 0 | QOS: **QOS_VALUE_INVALID** | TOPIC_NAME: $iothub/twin/res/204/?$rid=5&$version=102 | PAYLOAD_LEN: 0

The code works and I could look the result on IoTCentral PnP without problem are only the log messages not correct (for what I could see)

changing the files in the following way resolve the problem
If I had not done errors on my code... in this case sorry for that

$ git diff
diff --git a/**digitaltwin_client/src/dt_client_core.c b**/digitaltwin_client/src/dt_client_core.c
index ceafe99..412ea76 100644
--- a/digitaltwin_client/src/dt_client_core.c
+++ b/digitaltwin_client/src/dt_client_core.c
@@ -84,7 +84,8 @@ 
```
static const size_t methodInternalErrorLen = sizeof(methodInternalError) - 1;

static const int httpBadRequestStatusCode = 400;

-MU_DEFINE_ENUM_STRINGS(DIGITALTWIN_CLIENT_RESULT, DIGITALTWIN_CLIENT_RESULT_VALUES);
+MU_DEFINE_ENUM_STRINGS_WITHOUT_INVALID(DIGITALTWIN_CLIENT_RESULT, DIGITALTWIN_CLIENT_RESULT_VALUES);
```
diff --git a/**umqtt/src/mqtt_client.c b**/umqtt/src/mqtt_client.c
index 35fdbb5..faab397 100644
--- a/umqtt/src/mqtt_client.c
+++ b/umqtt/src/mqtt_client.c
@@ -32,7 +32,8 @@
```
static const char* const TRUE_CONST = "true";
static const char* const FALSE_CONST = "false";

-MU_DEFINE_ENUM_STRINGS(QOS_VALUE, QOS_VALUE_VALUES);
+MU_DEFINE_ENUM_STRINGS_2(QOS_VALUE, QOS_VALUE_VALUES);

#define MQTT_STATUS_INITIAL_STATUS      0x0000
#define MQTT_STATUS_CLIENT_CONNECTED    0x0001
@@ -656,7 +657,8 @@ static void ProcessPublishMessage(MQTT_CLIENT* mqtt_client, uint8_t* initialPos,
         if (is_trace_enabled(mqtt_client))
         {
             trace_log = STRING_construct_sprintf("PUBLISH | IS_DUP: %s | RETAIN: %d | QOS: %s | TOPIC_NAME: %s", isDuplicateMsg ? TRUE_CONST : FALSE_CONST,

             -isRetainMsg ? 1 : 0, MU_ENUM_TO_STRING(QOS_VALUE, qosValue), topicName);
             +isRetainMsg ? 1 : 0, MU_ENUM_TO_STRING_2(QOS_VALUE, qosValue), topicName);
         }
```
#endif
         uint16_t packetId = 0;

**Comment by jasmineymlo** [comment]
Close as this should be fixed by #1200.

**Comment by az-iot-builder-01** [comment]
@PezzoniL, @jasmineymlo, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
